### PR TITLE
🐛 Update linters and fix reported issues

### PR DESCRIPTION
**Pull request by Fedosin**
**What this PR does / why we need it**:

This PR enables additional linters and removes unused (commented) ones. All issues reported by linters are resolved either with "make lint-fix" or manually.

Noticeably, we fix implicit memory aliasing in for loops. In golang the iteration variable is reused. At each iteration, the value of the next element in the range expression is assigned to the iteration variable; "provider" doesn't change, only its value changes. Hence, the expression &provider is referring to the same location in memory.

The recommended solution for the issue is to redeclare "provider" inside the for loop.

For more information: https://github.com/golang/go/discussions/56010

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -141,8 +147,9 @@ issues:
     # Exclude some linters from running on tests files.
     - path: _test\.go
       linters:
-        - gocyclo
-        - dupl
         - gosec
-        - gochecknoglobals
-        - goerr113
+    - path: envtest/environment.go
```
Comment (furkatgofurov7) [inline .golangci.yaml:151]: where is this located?

File: cmd/main.go [hunk 2]
```diff
@@ -130,7 +128,12 @@ func main() {
 		klog.Infof("Profiler listening for requests at %s", profilerAddress)
 
 		go func() {
-			klog.Info(http.ListenAndServe(profilerAddress, nil))
+			server := &http.Server{
+				Addr:              profilerAddress,
+				ReadHeaderTimeout: 3 * time.Second,
+			}
```
Comment (furkatgofurov7) [inline cmd/main.go:134]: interesting, was this also reported as issue in linter? How are we coming up with 3 seconds for `ReadHeaderTimeout`

**Review by furkatgofurov7 — COMMENTED** [review]
Thanks for improvement, overall looks good, 2 questions inline:

File: .golangci.yaml (hunk 1 above)
Comment (alexander-demicev) [reply .golangci.yaml:151]: it's now in `internal` package

Comment (furkatgofurov7) [reply .golangci.yaml:151]: ah yes so it has to be: `internal/envtest/environment.go`

Comment (Fedosin) [reply .golangci.yaml:151]: fixed :)

File: cmd/main.go (hunk 2 above)
Comment (Fedosin) [reply cmd/main.go:134]: yep, it was reported by `gosec` linter. I thought that 3 second timeout is sane...

**Review by alexander-demicev — APPROVED** [review]
/approve

**Pull request merged**